### PR TITLE
Add build go:1.22

### PR DESCRIPTION
**Pull request by dohernandez**
Update golang version to the latest 1.22

### CODE REVIEW

File: golang/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-ARG VERSION="1.21"
+ARG VERSION="1.22"
```
Comment (pchico83) [inline golang/Dockerfile:1]: Could you add 1.22 in this file?
https://github.com/okteto/devenv/blob/main/docker-compose.golang.yml#L12

**Review by pchico83 — APPROVED** [review]

**Comment by pchico83** [comment]
@dohernandez thanks for your contribution!

**Comment by pchico83** [comment]
@dohernandez build is failing on CI, were you able to build it locally?
```
   9 |     
  10 | >>> RUN go install github.com/codegangsta/gin@latest && \
  11 | >>>     go install github.com/go-delve/delve/cmd/dlv@latest && \
  12 | >>>     go install golang.org/x/tools/gopls@latest && \
  13 | >>>     curl -sSfL https://raw.githubusercontent.com/cosmtrek/air/master/install.sh | sh -s -- -b /usr/bin -d v1.42.0
  14 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c go install github.com/codegangsta/gin@latest &&     go install github.com/go-delve/delve/cmd/dlv@latest &&     go install golang.org/x/tools/gopls@latest &&     curl -sSfL https://raw.githubusercontent.com/cosmtrek/air/master/install.sh | sh -s -- -b /usr/bin -d v1.42.0" did not complete successfully: exit code: 2
```

**Comment by dohernandez** [comment]
@pchico83 

Yes, I was able to run it locally but, half successful. 

When I run the command

```
docker buildx bake -f docker-compose.golang.yml
```
and `docker-compose.golang.yml` has all the actual services defined, it failed too, but if I comment out `golang1.18` service then the build passes.

**Comment by pchico83** [comment]
> @pchico83
> 
> Yes, I was able to run it locally but, half successful.
> 
> When I run the command
> 
> ```
> docker buildx bake -f docker-compose.golang.yml
> ```
> 
> and `docker-compose.golang.yml` has all the actual services defined, it failed too, but if I comment out `golang1.18` service then the build passes.

@dohernandez let's remove 1.18 then. The image would still be in dockerhub, we would just stop rebuilding it

**Comment by pchico83** [comment]
Merging PR, build failure is unrelated to this change. We will look into it on a different PT

**Pull request merged**